src/main.rs

```diff
@@ -7,36 +7,47 @@ use dirs::config_dir;
 use discord_rpc_client::Client as DiscordClient;
 use mpd::{Client as MPDClient, Song, State};
 use regex::{Captures, Regex};
-use toml::Value;
+use serde::{Deserialize, Serialize};
 
 const IDLE_TIME: u64 = 5;
 const ACTIVE_TIME: u64 = 1;
 
-const DISCORD_ID: &str = "677226551607033903";
+const DISCORD_ID: u64 = 677226551607033903;
 const DEFAULT_HOST: &str = "localhost:6600";
 const DETAILS_FORMAT: &str = "$title";
 const STATE_FORMAT: &str = "$artist / $album";
 
+#[derive(Serialize, Deserialize)]
+struct Format {
+    details: String,
+    state: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Config {
+    id: u64,
+    hosts: Vec<String>,
+    format: Option<Format>,
+}
+
 /// Creates the config directory and default configuration file
 fn create_config(path: &Path, filename: &str) -> std::io::Result<()> {
     println!("creating directory at '{:?}'", path);
     fs::create_dir_all(path)?;
 
     println!("creating default config file");
-    let mut config = fs::File::create(path.join(filename))?;
-    config.write_all(
-        format!(
-            "id = {}
-hosts = ['{}']
-
-[format]
-details = \"{}\"
-state = \"{}\"
-    ",
-            DISCORD_ID, DEFAULT_HOST, DETAILS_FORMAT, STATE_FORMAT
-        )
-        .as_bytes(),
-    )?;
+    let mut config_file = fs::File::create(path.join(filename))?;
+
+    let config = Config {
+        id: DISCORD_ID,
+        hosts: [DEFAULT_HOST.to_string()].to_vec(),
+        format: Some(Format {
+            details: DETAILS_FORMAT.to_string(),
+            state: STATE_FORMAT.to_string(),
+        }),
+    };
+
+    config_file.write_all(toml::to_string(&config).unwrap().as_bytes())?;
```

Comment:
Think that if you remove the `?;` at the end, you can remove the `Ok(())` under and just have this be the last line.

Reply:
Can indeed, sorted. Cheers.